http/obmc_shell.hpp

```diff
@@ -58,45 +58,15 @@ class Handler : public std::enable_shared_from_this<Handler>
         {
             // CHILD
 
-            // sets the effective user ID
-            // as service user
-            bool isUidSet = false;
             if (auto userName = session->getUserName(); !userName.empty())
             {
-                if (struct passwd* pw = getpwnam(userName.c_str());
-                    pw != nullptr)
-                {
-                    int uidr = setuid(pw->pw_uid);
-                    if (!uidr)
-                    {
-                        isUidSet = true;
-                    }
-                    else
-                    {
-                        BMCWEB_LOG_ERROR << "sets service user failed Error: "
-                                         << uidr;
-                    }
-                }
-                else
-                {
-                    BMCWEB_LOG_ERROR << "getpwnam return null passwd";
-                }
+                // replaces the current process with /bin/login chiled process
+                execl("/bin/login", "/bin/login", "-f", userName.c_str(), NULL);
```

Comment:
execl() can fail and you must handle the failure.

Reply:
done